doc/manual/building_developing.rst

```diff
@@ -0,0 +1,240 @@
+
+    to see the list of suitable build targets directly.
+
+Any of these commands should produce a directory ``result`` which contains a file ``boot.bin``. As described in :ref:`writing-flash`, if your core device is currently accessible over the network, it can be flashed with ``artiq_coremgmt``. If it is not connected to the network:
+
+1. Power off the board, extract the SD card and load ``boot.bin`` onto it manually.
+2. Insert the SD card back into the board.
+3. Ensure that the DIP switches (labeled BOOT MODE) are set correctly, to SD.
+4. Power the board back on.
+
+Optionally, the SD card may also be loaded at the same time with an additional file ``config.txt``, which can contain preset configuration values in the format ``key=value``, one per line. The keys are those used with ``artiq_coremgmt``. This allows e.g. presetting an IP address and any other configuration information.
+
+After a successful boot, the "FPGA DONE" light should be illuminated and the board should respond to ping when plugged into Ethernet.
+
+Booting over JTAG/Ethernet
+""""""""""""""""""""""""""
+
+It is also possible to flash/boot Zynq devices over USB and Ethernet. Flip the DIP switches to JTAG. The scripts ``remote_run.sh`` and ``local_run.sh`` in the ARTIQ-Zynq repository, intended for use with a remote JTAG server or a local connection to the core device respectively, were written at M-Labs to accomplish this. Both make use of the netboot tool ``artiq_netboot``, see also its source `here <https://git.m-labs.hk/M-Labs/artiq-netboot>`__, which is included in the ARTIQ-Zynq development environment. Adapt the relevant script to your system or read it closely to understand the options and the commands being run; note for example that ``remote_run.sh`` as written only supports ZC706.
+
+You will need to generate the gateware and firmware first, either through ``nix build`` or incrementally as below. For an incremental build add the option ``-i`` when running either of the scripts.
+
+.. warning::
+
+    A known Xilinx hardware bug on Zynq prevents repeatedly loading the SZL bootloader over JTAG (i.e. repeated calls of the ``*_run.sh`` scripts) without a POR reset. If necessary, physically set a jumper on the ``PS_POR_B`` pins of your board and use the M-Labs `POR reset script <https://git.m-labs.hk/M-Labs/zynq-rs/src/branch/master/kasli_soc_por.py>`_.
+
+.. _zynq-jtag-boot :
+
+Zynq incremental build
+^^^^^^^^^^^^^^^^^^^^^^
+
+In some circumstances, especially if you are developing ARTIQ, you may prefer to construct ``boot.bin`` manually. This is a little more involved, as ``boot.bin`` is the combination of several files, specifically ``szl.elf`` (an open-source bootloader for Zynq `written by M-Labs <https://git.m-labs.hk/M-Labs/zynq-rs/src/branch/master/szl>`_, used rather than Xilinx's FSBL), ``runtime`` or ``satman`` (the firmware) and ``top.bit`` (the gateware).
```

Comment:
Slightly confused by what changes this is asking for -- to rephrase this not in terms of constructing boot.bin? Halfway through the instructions (line 222) there's a note to stop there if using netboot, since boot.bin isn't necessary for that.